### ISSUE TITLE
Screen reader: Alt+Cursor in Suggest Widget reads line two times

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/vscode#4352]
- VSCode Version: 0.10.12-alpha
- OS Version: Windows 10 / NVDA screen reader

Steps to Reproduce:
1. enable NVDA
2. open TS file
3. open code complete
4. Press Alt+Cursor+Down to navigate to next entry

Observe: screen reader reads entry two times (at least that is what I see in the Speech Viewer)

![capture](https://cloud.githubusercontent.com/assets/1931590/13841029/b0d96494-ec26-11e5-9e68-2b47abad8e43.png)

**Comment by alexandrudima** [comment]
This is unfortunate, but reflects reality.

The first time you choose an item, it is rendered without the details icon. The widget fires a request to the language service to ask if the item has details and then it repaints to show that it has details.

The aria alerts reflect what the widget paints. Most of the times, this request is so fast that you don't notice the flicker visually.

IMHO the data structures are flawed, they should have a boolean up front for each item that contains details.

**Comment by jrieken** [comment]
For some language services the extra info is a second (and more expensive call) and the detail information isn't only loaded on the second Ctrl+Space but on focus (it shows more info). Since the backend cannot blindly fetch details for each and every item and since the backend doesn't know what item the UI will focus first it's not that easy.

**Comment by dbaeumer** [comment]
@alexandrudima assigning to you that it doesn't get lost.

Doesn't the screen reader reads the sentence on focus change in the complete widget?

**Comment by alexandrudima** [comment]
@jrieken My suggestion is simple. The suggestions should contain a boolean field if they have details or not.

The backend does not need to fetch details for each and every item, but say that it has such details. In many ASTs I suppose that would be an if stmt on the node?

If a backend really does not know then it should say it has details for all suggestions and then fail when asked for those details (as it works today).

**Comment by jrieken** [comment]
that's how it works. they never really know if there more data or not unless the provider doesn't offer the resolveDetails call - I can reflect that in the API

**Comment by jrieken** [comment]
We now only register the details-resolver when the provider actually supports it. I don't see the double request anymore. Closing.
